### DISCUSSION TITLE
New project domain www.hardinfo2.org

### DISCUSSION THREAD

**Discussion by hwspeedy** [hardinfo2/hardinfo2#3] [Announcements]
**hardinfo2 domain is alive**

Would have loved to stayed on hardinfo.org, but it was not fair to ask for as the domain is personal and is used for personal stuff, sorry. 

So the community project will be branded as **hardinfo2** with binary executable hardinfo2, github organisation hardinfo2, etc.

**lets go - who wants to make a better webpage?** - the bar is really low right now :)

Project page: https://www.hardinfo2.org

Development - pulled every hour from github:
https://github.com/hardinfo2/server

PS: I will be sharing codes for the domain with 2 others, as soon as the organization is build up, so we can be sure this domain is always part of hardinfo2 project.

**Comment by mckaygerhard** [comment]
hey do you buy a domain? ok good but i dont like it! well i try to explain my selft: you will mantiain and then if something happened to you we will have troubles.. do you understand me? that is a hard powerfull reason why i tel you to made an org (hardinfo2) event just a repo in your userspace..

**Reply by hwspeedy** [reply]
I think you need to read my messages again, please.
 - I have invited the two highest contributors to be co maintainers - If they decide not to - I will invite others to be co maintainers, so that there is always 3 - The developers will have full write access to repo. This is the most open community edition (Which is my vision of FOSS) - The DNA of open source - we develop together.
- As written above I will share the codes for the domain - this means that anyone of the 3 persons can choose to pay for the domain, can change the domain and even transfer the domain. But first we need the organization in place.

The survival of the project is important to everyone and I think this should be in the right direction.

**Comment by mckaygerhard** [comment]
mostly due the hardinfo2 need a hard dependency on the server storage for .. https://github.com/hardinfo2/server . so i guess its better a domain based in somethig perpetual like a github (i hate githsit cos is M$) .. 

PS: do you noted that an org was the right choose? @hwspeedy  you have the server and the hardinfo2 repos in the same place pretty organized! :+1:

**Reply by hwspeedy** [reply]
Nope - does not work with github.

The domain is in place and a server is in place. I still need to do the initial setup for DB - but already you can submit to the server repo and the project homepage is updated within 1 hour. 
The homepage will be the user entry page for the hardinfo2 project and also where we try to invite people to join the development here on github - provided to us free of charge by M$, which support open source to improve their image - better than nothing and should be acknowledged.

**Reply by mckaygerhard** [reply]
> . I still need to do the initial setup for DB

i mean a nextjs server.. something that can runs deployed in github.. i harte any javascript crap but this is perfect for.. well it was only an idea.. also any free service .. or something independient..

**Comment by mckaygerhard** [comment]
> **lets go - who wants to make a better webpage?** - the bar is really low right now :)

what about make a conquest?

**Reply by hwspeedy** [reply]
Great idea if you mean contest - but first we need at least 1 contestant with an idea of how the page should look. The development will then be done as a group effort.

I will be focusing on the release of hardinfo2 app - please all feel free to help - see prerelease for more info, thanx.

**Reply by mckaygerhard** [reply]
pifff .. i already requested to @diazvictor  XD :eyeglasses:

**Comment by diazvictor** [comment]
> **lets go - who wants to make a better webpage?** - the bar is really low right now :)

I am interested in collaborating with the web.

**Reply by hwspeedy** [reply]
That sound really great - I have added you to the project.

Would you be able to implement JSON->MariaDB and MariaDB->JSON - this is the missing part before changing the webpage.

I will setup the initial database in MariaDB and submit info in the server repo along with the json format.

**Comment by mckaygerhard** [comment]
@hwspeedy  sorry about the issue opened in hardinfo2 repo.. yes you are right!  well @diazvictor  knows a person close to him that can package and promote the hardinfo2 to official alpine project.. its just time to happened.. 

1. about web server.. i thinks LAMP is a old and unconfortable way to deploy a simple retrieve api from the past days.. today's a nextjs is a better way and shlud work in places such like gitlab pages, guthub deploy actions and any web server that dont need php engine.. just static generation and backend using javascript (i HATE javascript but well fucking stupid tech) .. sorry is just an idea.. 
2. about the database.. mariadb is another complication layer.. so mongodb provides free access and storage around terabytes . so is a realy obious options in the line of modernity.. as you pointed like the gtk2/3 case.. 

in resume.. we shiould change the older past days LAMP overcomplicated (that forced a place to have an php engine and database layer) 

towards a more modern environment without a complicated database layer that involves storing data in the cloud, taking advantage of the services that these exploitative companies provide for free @diazvictor  and @hwspeedy

**Reply by hwspeedy** [reply]
**Packaging**
The reason I have put the packaging into the project is to get a common ground for developers and distro maintainers - this has been a problem in several ways. 
Also the many users picking from GitHub - due to missing releases - needs more guidance and a package to install is way better.

**Choice of Webserver and Database**
It is about KISS (Keep It Simple Stupid) 

**76.5% of all web server are run by LAMP.**

Therefor we get the most developers and any green person can have fun on this project by using LAMP, which will be beneficial to the persons learning process.

**Regarding Hosting**
The project will be paying for its own metal
We want to be sure to have no surprises and want to do FOSS server development via GitHub.

**Reply by diazvictor** [reply]
I have no problems, for the moment I will take care of the FrontEnd and build a template. https://github.com/hardinfo2/server/pull/2

**Reply by mckaygerhard** [reply]
looks good to me.. the project already runs in lamp ok :+1:  , but in future.. as i know current modern development dont use LAMP anymore, we have nextjs, react, vuejs, golang deploys, etc.. we should move to modernity in next years

technically speaking.. usage of RDBMS for a two tables is waste and unoptimal, mongodb is the right choose.. due the large xperience in large amount of data

**Reply by hwspeedy** [reply]
@mckaygerhard 
You are mixing frontend and backend.
The server will use LAMP for a very long time. This project is a small amount of data.  The project will be using standard well proven setup.

**Please stop propaganda and do some work if you like to help the project, thanx.**

**Reply by mckaygerhard** [reply]
ok @hwspeedy  but propaganda: sorry gut: front: its only the web page, that is already doing by @diazvictor  that started a good thing, 

but backend: LAMP is on deprecation.. you talked about modernity: today's tech are mongodb for only two tables and mostly if a small amouont of data.. deploy mysql is "too much work" for.. 

if i will do some contribution will be on future to change that! also provide support for older people.. i really hate that majorities supress minorities cos i was on minorities and that is not fair.. so if you noted all my contributions are always to suppor tthose people that not have voice or does not have enought support!

be patient.. you do a great work! but we have a life to run outside of the screens :+1:

**Reply by hwspeedy** [reply]
@mckaygerhard 

**You are continuing propaganda:** 
Linux - still very actively maintained
Apache2 - still very actively maintained
MariaDb - still very actively maintained
PHP - still very actively maintained

**So LAMP is in no way deprecated! - it is the most used  - and a very good base for 90% of developers**

This is your personal vendetta because GTK2 is not maintained and therefor dead.

Please keep it together, thanx. 

PS: Propaganda are statements, that are not factually true - In FOSS community development, **work** contributions are the most helpful!
